dr_trigger.py

```diff
@@ -42,7 +71,9 @@ def trigger_dump(s_pid, s_port, path):
         sys.exit(1)
 
     sock = connect_to_server(server_pid)
-    request_dump(sock, port, file.fileno())
+    if sock:
+        request_dump(sock, port, file.fileno())
+
     file.close() 
     sock.close()
     return path 
```

Comment:
You are returning path even if you didn't connect to server. in which case the file is empty.